File: max-subarr.c
#include <stdio.h>

int maxSubArray(int* nums, int numsSize);

int main(void)
{
    int numbers[] = {-2, -3, -1};
    int size = sizeof(numbers)/sizeof(numbers[0]);
    printf("%i\n", maxSubArray(numbers, size));
}

int maxSubArray(int* nums, int numsSize)
{
    int curr_sum = 0;
    int best_sum = nums[0];
    for (int i = 0; i < numsSize; i++)
    {
        curr_sum += nums[i];
        if (best_sum < curr_sum)
            best_sum = curr_sum;
        if (curr_sum < 0)
            curr_sum = 0;
    }
    return best_sum;
}

File: max-subarr.py
def max_subarray(nums):
    """Find the largest sum of any contiguous subarray."""
    current_sum = 0
    best_sum = nums[0]
    for x in nums:
        current_sum += x
        if best_sum < current_sum:
            best_sum = current_sum
        if current_sum < 0:
            current_sum = 0
    return best_sum

print(max_subarray([-1, -3]))